File: app/src/main/java/com/example/testeglobojeremias/VideoViewModelFactory.java
package com.example.testeglobojeremias;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class VideoViewModelFactory implements ViewModelProvider.Factory {

    private final Application application;
    private final long movieID;

    public VideoViewModelFactory(Application application, long movieID) {
        this.application = application;
        this.movieID = movieID;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new VideosViewModel(application, movieID);
    }
}

File: app/src/main/java/com/tiagopereira/globotmdb/viewmodel/factory/VideoViewModelFactory.kt
package com.tiagopereira.globotmdb.viewmodel.factory

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.tiagopereira.globotmdb.viewmodel.VideoViewModel
import com.tiagopereira.globotmdb.viewmodel.repository.VideoRepository

class VideoViewModelFactory constructor(private val repository: VideoRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return if (modelClass.isAssignableFrom(VideoViewModel::class.java)) {
            VideoViewModel(this.repository) as T
        } else {
            throw IllegalArgumentException("ViewModel Not Found")
        }
    }
}